### PR TITLE
Add let keyword doc

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #34601.

r? @rust-lang/docs

### CODE REVIEW

**Comment by kennytm** [comment]
How will this appear in the doc and search? (I hope it won't say "*let (primitive)*" 😄)

**Comment by QuietMisdreavus** [comment]
It'll say "`let` (keyword)" - we already have the mechanism in place to give keywords their own section and signifiers. Here's how `fn` looks in today's nightly docs:

image

image

image

File: src/libstd/keyword_docs.rs [hunk 1]
```diff
@@ -26,3 +26,35 @@
...
+/// ```
+///
+/// By default, all variables are **not** mutable. If you want a mutable variable,
+/// you'll have to use the `mut` keyword.
+///
+/// Example:
+///
+/// ```rust
+/// # #![allow(unused_assignments)] fn main() {
+/// let mut x = 3; // We create a mutable variable named `x` with the value `3`.
```
Comment (QuietMisdreavus) [inline src/libstd/keyword_docs.rs]: You don't have to include a `fn main` to apply these crate-level lints. Rustdoc will already move them outside its generated `main` function.

Comment (GuillaumeGomez) [reply src/libstd/keyword_docs.rs]: Oh cool! Completely forgot about it. :)

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**